### DISCUSSION TITLE
Handling streaming effects

### DISCUSSION THREAD

**Discussion by RenateEilers** [haskell-effectful/effectful#128] [Q&A] [answered]
We are trying to create an effect that returns a [Streamly](https://hackage.haskell.org/package/streamly-0.8.3)  stream, which is of type `t m a`, where `t` represents the stream, and `m` is the monad in which the result of type `a` is returned. We would like to create an effect `StreamEffect :: Effect` with a constructor `StreamFromDB :: ... (relevant parameters) ... -> StreamEffect m' (t m a)`. We would like to be generic in `m`, since we need to have both pure and impure interpreters. Is there a way to go about this without threading the `m` as a type parameter. That is, without having `StreamEffect m`, and if another effect uses `StreamEffect` it needs to be supplied with the type parameter as well: `OtherEffect m :: Effect`. 

Thanks in advance!

**Comment by ocharles** [comment]
It's generally a lot easier to anwser these questions with a concrete example. Would you be able to provide an example of what is you want, ideally using terminology from the domain you're working in?

**Comment by arybczak** [comment, marked as answer]
As @ocharles pointed out, I'm not sure what you want to do without seeing a concrete example, but perhaps #108 will help. I think it was a similar problem, but with `conduit`.

**Reply by RenateEilers** [reply]
This helped, thanks a bunch!
